packages/backend/package.json

```diff
@@ -23,6 +23,7 @@
     "@withorbit/api": "0.0.1",
     "@withorbit/core": "0.0.1",
     "@withorbit/firebase-support": "0.0.1",
+    "ajv": "^8.5.0",
```

Comment:
Ah, yes. I've run into a few of these issues before. This being a monorepo makes it more confusing, but the underlying situation is not actually all that different from the troubles of a normal Node project in which you depend on `a` and `b@2.0`, while `a` internally depends on `b@1.0`. The Node resolver expects to resolve the same package for `require` calls in your module as well as in `a`: you've got to pick just one to load.

In such situations, you must either use the old version of `b` yourself until `a` upgrades, or force `a` to use `b@2.0` via [selective resolutions](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/). The latter may or may not work, depending on the nature of the breaking change. In practice, upgrading `a` sometimes suffices, since they might have updated their dependency on `b`.

This solution amounts to the latter: by installing the newer version of `ajv` in `backend/node_modules`, Node will preferentially resolve that one over the one at the monorepo root. This works because it happens that aside from `eslint`, none of `backend`'s dependencies depend on an older version of `ajv`. And it happens to work when running tests in `api` because none of the dependencies of that module depend on an older version of `ajv` either. We're really just getting lucky there, I guess!

Can we do better? Well... in our project, we have _many_ libraries which depend on `ajv` 6.x:
```
[ui (master)]$ yarn why ajv
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "ajv"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "ajv@6.12.6"
info Has been hoisted to "ajv"
info Reasons this module exists
   - "workspace-aggregator-e8a52493-a98e-40c9-a67b-5c8d3ba1a303" depends on it
   - Hoisted from "_project_#eslint#ajv"
   - Hoisted from "_project_#webpack#ajv"
   - Hoisted from "_project_#schema-utils#ajv"
   - Hoisted from "_project_#terser-webpack-plugin#schema-utils#ajv"
   - Hoisted from "_project_#webpack#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#backend#eslint#ajv"
   - Hoisted from "_project_#@withorbit#ui#eslint#ajv"
   - Hoisted from "_project_#eslint#table#ajv"
   - Hoisted from "_project_#eslint#@eslint#eslintrc#ajv"
   - Hoisted from "_project_#@withorbit#ui#react-docgen-typescript-loader#@webpack-contrib#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#app#@expo#webpack-config#webpack#ajv"
   - Hoisted from "_project_#@withorbit#app#webpack#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#web-component#webpack#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#app#@expo#webpack-config#worker-loader#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#app#@expo#webpack-config#mini-css-extract-plugin#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#app#@expo#webpack-config#webpack#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#app#@expo#webpack-config#babel-loader#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#app#@expo#webpack-config#terser-webpack-plugin#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#app#@expo#webpack-config#url-loader#schema-utils#ajv"
   - Hoisted from "_project_#@withorbit#ui#@storybook#react#@storybook#core#postcss-loader#schema-utils#ajv"
info Disk size without dependencies: "1.12MB"
info Disk size with unique dependencies: "1.86MB"
info Disk size with transitive dependencies: "1.9MB"
info Number of shared dependencies: 5
=> Found "@withorbit/api#ajv@8.4.0"
info This module exists because "_project_#@withorbit#api" depends on it.
=> Found "exegesis#ajv@6.12.5"
info This module exists because "_project_#@withorbit#app#firebase-tools#exegesis-express#exegesis" depends on it.
info Disk size without dependencies: "1.12MB"
info Disk size with unique dependencies: "1.86MB"
info Disk size with transitive dependencies: "1.9MB"
info Number of shared dependencies: 5
=> Found "har-validator#ajv@6.12.5"
info This module exists because "_project_#@withorbit#firebase-support#@firebase#rules-unit-testing#request#har-validator" depends on it.
info Disk size without dependencies: "1.12MB"
info Disk size with unique dependencies: "1.86MB"
info Disk size with transitive dependencies: "1.9MB"
info Number of shared dependencies: 5
```

And I see that `eslint@master` [still references](https://github.com/eslint/eslint/blob/master/package.json) `ajv` 6.x, so updating our dependencies won't fix the issue. I guess we're stuck! There's an issue about updating in eslint [here](https://github.com/eslint/eslint/issues/13888). (Boy, some of the comments from the `ajv` people about the churn they're dealing with in the JSON Schema spec scare me!)

The only potential improvement I see would be to declare our intent somewhat more explicitly here, so that we make clear that `backend` doesn't directly depend on `ajv`: it just needs the version resolved a certain way. You should be able to do that by declaring a selective resolution instead of an explicit dependency, i.e.:

```
"resolutions": {
  "api/ajv": "^8.5.0"
}
```

Reply:
Oh wow, TIL. Thank you for the detailed explanation. That makes a lot more sense now. I will go ahead and add the resolutions to the `package.json`.

> (some of the comments from the ajv people about the churn they're dealing with in the JSON Schema spec scare me!)

Yeah same... I hope by keeping our spec relatively "vanilla" (i.e. no custom keywords or plugins) spec changes won't be breaking 🤞🏻. For better or worse `typescript-json-schema` keeps us rather constricted from getting ahead of ourselves.